src/NUnitFramework/framework/Is.cs

```diff
@@ -373,6 +373,19 @@ public static CollectionEquivalentConstraint EquivalentTo(IEnumerable expected)
 
         #endregion
 
+        #region ContainedIn
+
+        /// <summary>
+        /// Returns a constraint that tests whether the actual value
+        /// contains a value from the collection supplied as an argument.
+        /// </summary>
+        public static CollectionContainsAnyConstraint ContainedIn(IEnumerable expected)
```

Comment:
Thanks for adding this. I often also use the `Contains` class for this sort of check too. Would it be possible to also expose a `Contains.All(IEnumerable)` function there too?

Reply:
Anytime - just want to clarify, do you use `Contains` with iteration to perform the checks you need? Just want to make sure we are on the same page since they perform different actions. 

The naming might have been a bit misleading here - `Contains.All(IEnumerable)` already exists but is instead named SupersetOf(IEnumerable) within this class.

Reply:
@tbenne10 Thanks for asking, I appreciate you coming back to this PR after all this time.

You make a good point. I may've mistyped my earlier suggestion in haste. I'd suggest we disregard my suggestion around `Contains.All()` in the interest of keeping this PR simple. Having thought it through, I agree with @OsirisTerje 's suggestion that it might make sense to expose your new constraint as a `Does.ContainAny()` method, which could then be easily inverted to be `Does.Not.ContainAny()` using the already-existing `Not` constraint.


I think my original comment came from a place of concern around ambiguity of the name `Is.ContainedIn` in relation to some other API operations when both arguments are collections. If I understand your PR right, the following will pass the first assertion but fail the second and third because they apply fundamentally different checks, though it might not be clear simply by reading them that this will happen. 

```csharp
var a = new[] { 1, 2, 3 };
var b = new[] { 2, 4 };

Assert.That(a, Is.ContainedIn(b));
Assert.That(a, Is.SubsetOf(b));
Assert.That(b, Contains.All(b));
```